src/core/Akka.Streams/Dsl/DelayFlow.cs

```diff
@@ -0,0 +1,179 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="DelayFlow.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2018 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using Akka.Annotations;
+using Akka.Streams.Implementation.Fusing;
+using Akka.Streams.Stage;
+
+namespace Akka.Streams.Dsl
+{
+    /// <summary>
+    /// Allows to manage delay and can be stateful to compute delay for any sequence of elements,
+    /// all elements go through <see cref="NextDelay(T)"/> updating state and returning delay for each element
+    /// </summary>
+    /// <typeparam name="T">type of element</typeparam>
+    public interface IDelayStrategy<T>
+    {
+        /// <summary>
+        /// Returns delay for ongoing <paramref name="elem"/>, <code>TimeSpan.Zero</code> means passing without delay
+        /// </summary>
+        /// <param name="elem">element</param>
+        /// <returns></returns>
+        TimeSpan NextDelay(T elem);
```

Comment:
We have agreed on using full names instead of shortcuts from the jvm, i.e. `element` instead of `elem` , it's not 100% rule especially vor local variables etc. but we should do it in the API, so please change.